B23 Ex03 Ronen 319047718 Ido 315942193/VehicleFactory.cs

```diff
@@ -117,19 +113,104 @@ private Vehicle getCarFromUserByType(eVehicleType i_Type)
         }
 
+        private static float getElectricCarPowerLeftFromUser()
+        {
+            Console.Write("Please insert the electric left");
+            float powerLeft = float.Parse(Console.ReadLine());
+            return powerLeft;
+        }
+
+        private static eNumOfCarDoors getNumOfDoorsFromUser()
+        {
+            if (Enum.TryParse(Console.ReadLine(), out eNumOfCarDoors numOfDoors))
+            {
+                if (!Enum.IsDefined(typeof(eNumOfCarDoors), numOfDoors))
+                {
+                    throw new ArgumentException("number of doors not listed on the menu");
+                }
+            }
+            else
+            {
+                throw new FormatException(Display.InvalidEnumParameter());
+            }
+
+            return numOfDoors;
+        }
+
+        private static bool isCarType(eVehicleType i_Type)
+        {
+            return i_Type == eVehicleType.ElectricCar || i_Type == eVehicleType.DieselCar;
+        }
+        private static bool isBikeType(eVehicleType i_Type)
+        {
+            return i_Type == eVehicleType.ElectricBike || i_Type == eVehicleType.DieselBike;
+        }
+        private static bool isTruckType(eVehicleType i_Type)
+        {
+            return i_Type == eVehicleType.Truck;
+        }
+
+        private static bool isElectricVehicle(eVehicleType i_Type)
+        {
+            return i_Type == eVehicleType.ElectricBike || i_Type == eVehicleType.ElectricCar;
+        }
+
+        private static int getEngineSizeFromUser()
+        {
+            Console.Write("Please insert the Engine size: ");
+            int engineSize = int.Parse(Console.ReadLine());
+            return engineSize;
+        }
+
+        private static float getCargoFromUser()
+        {
+            Console.Write("Please insert cargo size: ");
+            float cargoSize = float.Parse(Console.ReadLine());
+            return cargoSize;
+        }
+
+        private static float getFuelLeftFromUser()
+        {
+            Console.Write("Please insert the fuel left: ");
+            float powerLeft = float.Parse(Console.ReadLine());
+            return powerLeft;
+        }
+
+        private static bool isDieselVehicle(eVehicleType i_Type)
+        {
+            return i_Type == eVehicleType.DieselCar || i_Type == eVehicleType.DieselBike || i_Type == eVehicleType.Truck;
+        }
+
+        private static string getModelFromUser()
+        {
+            Console.Write("Please insert the model: ");
+            string model = Console.ReadLine();
+            return model;
+        }
+
         private eMotorcycleLicense getBikeLicense()
         {
-            Console.WriteLine("Please insert the bike license type\n1. A1\n2. A2\n3. AA\n4. B1");
-            Enum.TryParse(Console.ReadLine(), out eMotorcycleLicense o_bikeLicense);
+            Display.BikeLicense();
+
+            if (Enum.TryParse(Console.ReadLine(), out eMotorcycleLicense o_bikeLicense))
```

Comment:
Change o_bikeLicense to motorcycleLicense

Reply:
Done